### PR TITLE
Added Snackbar for Notification

### LINKED ISSUE #73: Indicate to users that their submissions have been sent
Right now the dialog just closes when a user adds a clinic or reports one. There's nothing that indicates that their request went through. We should add a small notification somewhere on the screen.

### LINKED ISSUE #103: Set up toast notifications
We want to have a consistent design/ experience for notifications and user status messages. 

- decide what framework to use
- implement it. 
- Decide on a style and position.  (top left, red for errors, black for default, etc.)

### PR DESCRIPTION
**Pull request by amar-prabhu-29**
## Resolves #103 Resolves #73
Added a Snackbar Component to show error/success status after completing a form.
### Props
**showSnackBar:** Boolean (true/false)
**snackBarText:** String (The text to be added to the snack bar)

### Events
**closeSnackBar:** Emitted when the user clicks Ok! on scnackbar

### How to test
1) Adding a clinic
2) Flagging a clinic 

### Screenshots
Capture

### CODE REVIEW

File: components/NotificationSnackBar.vue
Comment (ann-kilzer) [inline components/NotificationSnackBar.vue:5]: nit: Can we use the Material Colors? https://v15.vuetifyjs.com/en/framework/colors#material-colors

```suggestion
            <v-btn color="indigo darken-1" text v-bind="attrs" @click="() => {$emit('closeSnackbar')}">
```

**Review by ann-kilzer — COMMENTED** [review]
It looks nice and modular. Thanks for the quick addition. I'll defer to @Ermish for the rest of the review because he's knowledgeable about Toast and filed the original ticket.

File: components/NotificationSnackBar.vue
Comment (amar-prabhu-29) [reply components/NotificationSnackBar.vue:5]: I was looking to keep all the colors in accordance with the current design.

Any suggestion on colors is welcome.

File: pages/add-clinic.vue
Comment (Ermish) [inline pages/add-clinic.vue:120]: this should be part of the Vue component `data` for proper reactivity and follow vue local state management guidelines.
See: https://v3.vuejs.org/api/options-data.html

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:149]: since this is moving to `data`, scope it to snackbar, like `showSnackBar` and `snackBarText`.

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:151]: techincally this event isn't `closed` it's when the close button is clicked. We should update this to `closeBtnClicked` in the snackbar component and rename this event handler `handleSnackBarCloseBtnClicked`

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:177]: this will need to be updated when moving `snackBarData` to `data` to be reactive.

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:179]: this text should use i18n translations. `$("your translation here")`

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:189]: should also use translations

File: components/NotificationSnackBar.vue
Comment (Ermish) [inline components/NotificationSnackBar.vue:5]: see earlier comment about updating the event name


edit: actually.....
1) shouldn't v-snackbar already have a built in close button? No need to add a second. We should just get rid of this button + event IMO. 
2) There's no need to emit the button click honestly....the consumer only cares about the event where the snackbar `closed` which is already handled with `v-model="showSnackBar"`

File: pages/add-clinic.vue
Comment (Ermish) [inline pages/add-clinic.vue:148]: I like that you re-used the same notification instead of building two!

File: pages/add-clinic.vue
Comment (Ermish) [inline pages/add-clinic.vue:153]: this shouldn't be tied to the close event since we can't know when it was. 
Move this to after the snackbar is opened and add a delay  like this:

```
await new Promise((resolve) => setTimeout(resolve, 1500));
```

Note: `submitData()` will need to be `async`

**Review by Ermish — CHANGES_REQUESTED** [review]
Have a few changes that we need  to make to the design.

File: pages/add-clinic.vue
Comment (renderf0x) [reply pages/add-clinic.vue:148]: Although, should we remove that console.log before merging?

File: components/NotificationSnackBar.vue
Comment (amar-prabhu-29) [reply components/NotificationSnackBar.vue:5]: I emitted the event for redirections cases,
In case theres a redirection to the home page, i choose to do it after the Ok! button has been clicked. 
Let me know your thought...

File: pages/add-clinic.vue
Comment (amar-prabhu-29) [reply pages/add-clinic.vue:120]: Im sorry i thought i added it to the data component 
i lost track of the brackets opening and closing ill do the necessary changes

File: components/NotificationSnackBar.vue
Comment (Ermish) [reply components/NotificationSnackBar.vue:5]: your thinking makes sense, but the parent can use the `v-model` change for that. Then they can check if it was closed by a timeout OR a button click.

File: pages/add-clinic.vue
Comment (Ermish) [reply pages/add-clinic.vue:148]: @renderf0x I haven't been commenting on `console.logs` since we don't have a proper logging system yet and they aren't helping on the user side.
BUT I was going to use them as points of find+replace targets when we get client side logging and they aren't hurting anything right now. 💡

**Comment by theyokohamalife** [comment]
Hey @amar-prabhu-29! Are you able to make the requested changes? We'd really like to merge your contribution.

**Pull request closed without merging**
